rust_src/src/floatfns.rs

```diff
@@ -147,15 +176,220 @@ fn ftruncate(arg: LispObject) -> LispObject {
 /// Return the floating point number equal to ARG.
 #[lisp_fn]
 fn float(arg: LispObject) -> LispObject {
-    if !arg.is_number() {
+    if arg.is_float() {
+        arg
+    } else if let Some(n) = arg.as_fixnum() {
+        LispObject::from_float(n as EmacsDouble)
+    } else {
         unsafe {
             wrong_type_argument(Qnumberp, arg.to_raw());
         }
     }
+}
 
-    if arg.is_float() {
-        arg
+/// Copy sign of X2 to value of X1, and return the result.
+/// Cause an error if X1 or X2 is not a float.
+#[lisp_fn]
+fn copysign(x1: LispObject, x2: LispObject) -> LispObject {
+    let f1 = x1.as_float_or_error();
+    let f2 = x2.as_float_or_error();
+    if libm::signbit(f1) != libm::signbit(f2) {
+        LispObject::from_float(-f1)
+    } else {
+        x1
+    }
+}
+
+/// Get significand and exponent of a floating point number.
+/// Breaks the floating point number X into its binary significand SGNFCAND
+/// \(a floating point value between 0.5 (included) and 1.0 (excluded))
```

Comment:
I believe you don't actually need to escape the `(` here.

The Emacs convention for escaping parentheses is because something gets confused if you start a line with a `(` (I think it's either highlighting or indentation). The C code suffers from this:

```c
DEFUN ("frexp", Ffrexp, Sfrexp, 1, 1, 0,
       doc: /* Get significand and exponent of a floating point number.
Breaks the floating point number X into its binary significand SGNFCAND
\(a floating point value between 0.5 (included) and 1.0 (excluded))
and an integral exponent EXP for 2, such that:
```

but the Rust code doesn't, because the line starts with `//`.